src/cryptoadvance/specter/cli.py

```diff
@@ -0,0 +1,263 @@
+            os.kill(pid, signal.SIGTERM)
+            time.sleep(0.3)
+            try:
+                os.remove(pid_file)
+            except Exception:
+                pass
+        elif daemon:
+            if not force:
+                print(f"PID file \"{pid_file}\" already exists. Use --force to overwrite")
+                return
+            else:
+                os.remove(pid_file)
+        if stop:
+            return
+    else:
+        if stop or restart:
+            print(f"Can't find PID file \"{pid_file}\"")
+            if stop:
+                return
+
+    app = create_app()
+    app.app_context().push()
+    init_app(app, hwibridge=hwibridge)
+
+    # watch templates folder to reload when something changes
+    extra_dirs = ['templates']
+    extra_files = extra_dirs[:]
+    for extra_dir in extra_dirs:
+        for dirname, dirs, files in os.walk(extra_dir):
+            for filename in files:
+                filename = os.path.join(dirname, filename)
+                if os.path.isfile(filename):
+                    extra_files.append(filename)
+
+    # if port is not defined - get it from environment
+    if port is None:
+        port = int(os.getenv('PORT', 25441))
+    else:
+        port = int(port)
+
+    # certificates
+    if cert is None:
+        cert = os.getenv('CERT', None)
+    if key is None:
+        key = os.getenv('KEY', None)
+
+    protocol = "http"
+    kwargs = {
+        "host": host,
+        "port": port,
+        "extra_files": extra_files,
+    }
+    if cert is not None and key is not None:
+        cert = os.path.abspath(cert)
+        key = os.path.abspath(key)
+        kwargs["ssl_context"] = (cert, key)
+        protocol = "https"
+
+    if hwibridge:
+        app.logger.info(
+            "Running HWI Bridge mode, you can configure access \
+            to the API at: %s://%s:%d/hwi/settings"
+            % (protocol, host, port)
+        )
+
+    # debug is false by default
+    def run(debug=debug):
+        try:
+            app.controller = Controller.from_port()
+        except:
+            app.controller = None
+        try:
+            port = 5000  # default flask port
+            if 'port' in kwargs:
+                port = kwargs['port']
+            else:
+                kwargs['port'] = port
+            # if we have certificates
+            if "ssl_context" in kwargs:
+                tor_port = 443
+            else:
+                tor_port = 80
+            app.port = port
+            app.tor_port = tor_port
+            app.save_tor_address_to = toraddr_file
+            if debug and (tor or os.getenv('CONNECT_TOR') == 'True'):
+                print(
+                    '* Warning: Cannot use Tor in debug mode. Starting in production mode instead.'
+                )
+            if tor or os.getenv('CONNECT_TOR') == 'True':
+                try:
+                    app.tor_enabled = True
+                    tor_util.start_hidden_service(app)
```

Comment:
Here we should make sure to disable debug or Tor will not work if debug=True (and the message at line 138 won't be True...)
```suggestion
                    tor_util.start_hidden_service(app)
                    debug = False
```